script/wait_for_file.sh

```diff
@@ -0,0 +1,8 @@
+wait_for_file() {
+  local file="$1"; shift
+  local wait_seconds="${1:-10}"; shift # 10 seconds as default timeout
+
+  until test $((wait_seconds--)) -eq 0 -o -f "$file" ; do sleep 1; done
+
+  ((++wait_seconds))
```

Comment:
And I got confused by prefix / postfix `--` and `++`. So if timeout was reached, `wait_seconds` will actually be `-1`. Still, in shell 0 is true, and non-zero is false, so you have to negate the return value: 

```
[ "$wait_seconds" -ne -1]
```